velPaint

```diff
@@ -85,20 +89,17 @@ while (my $line = <>) {
 	chomp $line;
 	$line =~ s/\r//g;
 
-	($x, $y, $z, $e, $f) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) Z([^\s]+) E([^\s]+) F($targetSpeed)$/;
-	($x, $y, $e, $f) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) E([^\s]+) F($targetSpeed)$/ unless (defined $x);
-	($x, $y, $z, $e) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) Z([^\s]+) E([^\s]+)$/ unless (defined $x);
-	($x, $y, $e) =
-	 $line =~ /G1 X([^\s]+) Y([^\s]+) E([^\s]+)$/ unless (defined $x);
+	if($line =~ /G1 X(?<x>\S+) Y(?<y>\S+)(?: Z(?<z>\S+))? E(?<e>\S+)(?: F(?<f>$targetSpeed))?$/ ||
```

Comment:
Using the named capture feature and making some of the pieces optional, this group of regular expression combines into a smaller set. This was especially useful after adding the expressions for the slic3r change.